plugins/permission-common/src/PermissionClient.ts

```diff
@@ -144,17 +177,75 @@ export class PermissionClient implements PermissionEvaluator {
     itemSchema: z.ZodSchema<TResult>,
     options?: AuthorizeRequestOptions,
   ) {
-    if (!this.enabled) {
-      return queries.map(_ => ({ result: AuthorizeResult.ALLOW as const }));
-    }
-
     const request: PermissionMessageBatch<TQuery> = {
       items: queries.map(query => ({
         id: uuid.v4(),
         ...query,
       })),
     };
 
+    const parsedResponse = await this.makeRawRequest(
+      request,
+      itemSchema,
+      options,
+    );
+
+    const responsesById = parsedResponse.items.reduce((acc, r) => {
+      acc[r.id] = r;
+      return acc;
+    }, {} as Record<string, z.infer<typeof itemSchema>>);
+
+    return request.items.map(query => responsesById[query.id]);
+  }
+
+  private async makeBatchedRequest(
+    queries: AuthorizePermissionRequest[],
+    options?: AuthorizeRequestOptions,
+  ) {
+    const request: Record<string, BatchedAuthorizePermissionRequest> = {};
+
+    for (const query of queries) {
+      const { permission, resourceRef } = query;
+
+      if (isResourcePermission(permission)) {
+        request[permission.name] ||= {
+          permission,
+          resourceRefs: [],
+          id: uuid.v4(),
+        };
+
+        if (resourceRef) {
+          request[permission.name].resourceRefs?.push(resourceRef);
+        }
+      } else {
+        request[permission.name] ||= {
+          permission,
+          id: uuid.v4(),
+        };
+      }
+    }
+
+    const parsedResponse = await this.makeRawRequest(
+      { items: Object.values(request) },
+      authorizePermissionResponseBatchSchema,
+      options,
+    );
+
+    return queries.map(query => {
+      const { id } = request[query.permission.name];
+
+      const item = parsedResponse.items.find(i => i.id === id)!;
```

Comment:
Just noting that this is N^2 (iterates through the items for every query)

Reply:
it should have any impact given the limited set of items, but I've pushed a fix 👍